vslib/Makefile.am

```diff
@@ -1,9 +1,35 @@
 AM_CXXFLAGS = $(SAIINC) -I$(top_srcdir)/lib -I/usr/include/libnl3
 
+if USE_VPP
+AM_CXXFLAGS += -Ivpp -DUSE_VPP
+endif
+
+CFLAGS_COMMON= -fPIC -pipe -Wall -Wcast-align -Wcast-qual -Wconversion -Wdisabled-optimization -Werror -Wextra -Wfloat-equal -Wformat=2 -Wformat-nonliteral -Wformat-security -Wformat-y2k -Wimport -Winit-self -Wno-inline -Winvalid-pch -Wmissing-field-initializers -Wmissing-format-attribute -Wmissing-include-dirs -Wmissing-noreturn -Wno-aggregate-return -Wno-padded -Wno-switch-enum -Wno-unused-parameter -Wpacked -Wpointer-arith -Wredundant-decls -Wshadow -Wstack-protector -Wstrict-aliasing=3 -Wswitch -Wswitch-default -Wunreachable-code -Wunused -Wvariadic-macros -Wwrite-strings -Wno-switch-default -Wconversion -Wno-psabi -Wcast-align=strict
+
+# TODO those are currently disabled flags from C files untill all warnings will be fixed
+DISABLED_FLAGS= \
+			   -Wno-unused-function \
+			   -Wno-sign-conversion \
+			   -Wno-conversion \
+			   -Wno-unused-variable \
+			   -Wno-sign-compare \
+			   -Wno-shadow \
+			   -Wno-discarded-qualifiers \
+			   -Wno-redundant-decls \
+			   -Wno-cast-align \
+			   -Wno-pointer-sign \
+			   -Wno-stringop-truncation \
+			   -Wno-cast-qual
```

Comment:
@yue-fred-gao can you look at those warnings? i disabled those since *c files in vppxlate/*c developer produce a lot of warnings which are treated as errors by our convention, and for many of them, i don't know what was actual developer intention, and without any unittests provided on that i don't want to change any of this code

@yue-fred-gao will you be able to help to eliminate those warnings ?

also what is a purpose of vppxlate directory? is there any purpose of that on other repo? or this is specifically designed to for vpp in sairedis/buildimage ? is there specific requirement that this code needs to be in C and not cpp ?

Reply:
I created an issue (https://github.com/sonic-net/sonic-platform-vpp/issues/149) to track this. 
vppxlate is the vpp API adaptation layer. It is specifically for vpp. There is some history that I don't know but I guess vpp APIs are C code so C is used here.